src/main.rs

```diff
@@ -111,8 +220,16 @@ fn main() {
         std::process::exit(0);
     }
 
-    // SETUP phase
-    let mut pid_proc = Process::new(opts.pid).expect("Failed accessing process");
+    // Initialize the tracking process
+    let mut pid_proc = match TrackedProcess::try_from(&opts) {
+			Err(e) => { 
```

Comment:
Another way to shorten the program would be to make more use of the `?` operator for error handling as in most cases either just the error is printed or the function/program is exited

Reply:
Especially when also using `anyhow` lots of `unwraps()` can also be replaced